PopPUNK/network.py

```diff
@@ -67,29 +67,54 @@ def fetchNetwork(network_dir, model, refList, ref_graph = False,
     """
     # If a refined fit, may use just core or accessory distances
     dir_prefix = network_dir + "/" + os.path.basename(network_dir)
+    if use_gpu:
+
+        # load CUDA libraries
+        try:
+            import cugraph
+            import cudf
+        except ImportError as e:
+            sys.stderr.write("cugraph and cudf unavailable\n")
+            raise ImportError(e)
+
+        graph_suffix = '.csv.gz'
+    else:
+        graph_suffix = '.gt'
     if core_only and model.type == 'refine':
         model.slope = 0
-        network_file = dir_prefix + '_core_graph.gt'
+        network_file = dir_prefix + '_core_graph' + graph_suffix
         cluster_file = dir_prefix + '_core_clusters.csv'
     elif accessory_only and model.type == 'refine':
         model.slope = 1
-        network_file = dir_prefix + '_accessory_graph.gt'
+        network_file = dir_prefix + '_accessory_graph' + graph_suffix
         cluster_file = dir_prefix + '_accessory_clusters.csv'
     else:
-        if ref_graph and os.path.isfile(dir_prefix + '.refs_graph.gt'):
-            network_file = dir_prefix + '.refs_graph.gt'
+        if ref_graph and os.path.isfile(dir_prefix + '.refs_graph' + graph_suffix):
+            network_file = dir_prefix + '.refs_graph' + graph_suffix
         else:
-            network_file = dir_prefix + '_graph.gt'
+            network_file = dir_prefix + '_graph' + graph_suffix
         cluster_file = dir_prefix + '_clusters.csv'
         if core_only or accessory_only:
             sys.stderr.write("Can only do --core-only or --accessory-only fits from "
                              "a refined fit. Using the combined distances.\n")
 
-    genomeNetwork = gt.load_graph(network_file)
-    sys.stderr.write("Network loaded: " + str(len(list(genomeNetwork.vertices()))) + " samples\n")
+    if use_gpu:
+        G_df = cudf.read_csv(network_file, compression = 'gzip')
+        genomeNetwork = cugraph.Graph()
+        if 'weights' in G_df.columns:
+            G_df.columns = ['source','destination','weights']
+            genomeNetwork.from_cudf_edgelist(G_df, edge_attr='weights', renumber=False)
+        else:
+            G_df.columns = ['source','destination']
+            genomeNetwork.from_cudf_edgelist(G_df,renumber=False)
+        sys.stderr.write("Network loaded: " + str(genomeNetwork.number_of_vertices()) + " samples\n")
+    else:
+        genomeNetwork = gt.load_graph(network_file)
+        sys.stderr.write("Network loaded: " + str(len(list(genomeNetwork.vertices()))) + " samples\n")
 
     # Ensure all in dists are in final network
-    networkMissing = set(map(str,set(range(len(refList))).difference(list(genomeNetwork.vertices()))))
+    vertex_list = set(get_vertex_list(genomeNetwork, use_gpu = use_gpu))
+    networkMissing = set(set(range(len(refList))).difference(vertex_list))
```

Comment:
This is the same check as in `main()`. Do we need it in both places? Is it better just in this function?

Reply:
If both are needed, it needs to become a function instead

Reply:
Worth checking loading in case user specifies incompatible data sources. Made a function in bb2f9e1.